components/service/glean/src/main/java/mozilla/components/service/glean/debug/GleanDebugBroadcastReceiver.kt

```diff
@@ -0,0 +1,68 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean.debug
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import mozilla.components.service.glean.Glean
+import mozilla.components.support.base.log.logger.Logger
+
+/**
+ * A class to receive broadcast "intents" for debugging purposes.
+ *
+ * There following intents are supported:
+ *
+ *   - org.mozilla.components.service.glean.sendPing
+ *         Sends a ping immediately.  It has two charSequence parameters:
+ *             - pingName: The name of the ping
+ *             - docType: The docType to use in the URL (optional: if not provided,
+ *               equals pingName)
+ *
+ *   - org.mozilla.components.service.glean.logPings
+ *         Turns on logging of ping contents to logCat.
+ */
+class GleanDebugBroadcastReceiver : BroadcastReceiver() {
```

Comment:
This is lacking test coverage. Moreover, I'd favour extending `GleanDebugActivity` instead of introducing a new mechanism, unless there's a good reason to do so. Moreover, `GleanDebugActivityTest` already gives you the framework to do test coverage :)

Reply:
I agree it needs testing. I was hoping to get feedback on the general approach before doing testing.

Reply:
I like the idea of using a broadcast receiver here, are there any security implications if this is allowed in more than just debug though?

Reply:
I've updated this to just use the `Activity` we already had, instead of the broadcast intent.  The biggest practical difference is the Activity is directed at a specific application rather than all applications using glean, which is probably a good thing, though less convenient.